docs/proposals/21-ipv6-singlestack-local.md

```diff
@@ -0,0 +1,213 @@
+
+To bring Gardener closer to supporting these kinds of networking setups, this GEP focuses on supporting IPv6 single-stack networking in the local Gardener environment.
+This keeps things focused on the most important changes to the API and central components while neglecting infrastructure-specific quirks and other difficulties that will arise with dual-stack networking.
+
+Once this enhancement has been implemented and IPv6 single-stack networking in local shoots is a stable feature, further changes can be proposed to support IPv6 single-stack on cloud infrastructure or IPv4/IPv6 dual-stack networking.
+
+### Goals
+
+- augment the Gardener API to allow selecting either IPv4 or IPv6 single-stack networking
+  - define a contract that all components (including networking extensions) need to follow to support this configuration
+  - allow extending the API for dual-stack networking later on while keeping backward-/forward-compatibility
+- adapt all relevant components in the local Gardener environment to support this configuration
+- add e2e tests to prevent regressions and guarantee the stability of the feature (especially because nobody is running it in productive environments)
+
+### Non-Goals
+
+- support IPv4/IPv6 dual-stack networking
+- support IPv6 single-stack networking on cloud infrastructure
+- support changing the networking setup of shoots from IPv4 to IPv6 single-stack or vice-versa
+- propose changes to `NetworkConfig` APIs of networking extensions or their implementation
+- support the legacy VPN tunnel (`ReversedVPN` feature gate is disabled)
+
+## Proposal
+
+A new feature gate `IPv6SingleStack` is added to gardener-apiserver.
+The IPv6-related fields and values can only be used if the feature gate is enabled.
+The feature gate cannot be enabled if the `ReversedVPN` feature gate is disabled (the legacy VPN tunnel solution is [not supported by provider-local](https://github.com/gardener/gardener/blob/83de074f1bc1c009f92e97a08289340591377af6/docs/extensions/provider-local.md#current-limitations) anyway).
+
+### `Shoot` API
+
+The `Shoot.spec.networking` section is extended as follows:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: Shoot
+spec:
+  networking:
+    # ...
+    ipFamilies:
+    - IPv6
+    podsV6: dead:.../64
+    nodesV6: dead:.../64
+    servicesV6: dead:.../64
+```
+
+`ipFamilies` is the central setting for specifying the IP families used for shoot networking.
+This field is inspired by the `Service.spec.ipFamilies` field in Kubernetes ([doc](https://kubernetes.io/docs/concepts/services-networking/dual-stack/#services)).
+The only valid value and default is `["IPv4"]` (IPv4 single-stack).
+If the `IPv6SingleStack` feature gate is enabled, `["IPv6"]` can be specified to switch to IPv6 single-stack.
+Later on, `["IPv4","IPv6"]` or `["IPv6","IPv4"]` can be supported for dual-stack networking (ordering is relevant for Kubernetes configuration like the `--service-cluster-ip-range` flag).
```

Comment:
I get what you're saying. We simply added this information to provide a reason why we are proposing an array field instead of a string value field (e.g., `ipFamilies=IPv4,IPv6,DualStack`)